File: Professional Image Modifier 1/[VB]-Professional Image Modifier 1/VB/ImageFunctions/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace ImageFunctions
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()

		End Sub




		Private Sub btnLoadNewImage_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLoadNewImage.Click
			Dim dr As DialogResult = openFile.ShowDialog()
			If dr = System.Windows.Forms.DialogResult.OK Then
				For Each file As String In openFile.FileNames
					listBox1.Items.Add(file)
					pbImage.Image = Image.FromFile(file)
				Next file
			End If
		End Sub

		Private Sub listBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles listBox1.SelectedIndexChanged
			If listBox1.SelectedIndex > ilDefault.Images.Count - 1 Then ' Must be added by the user
				pbImage.Image = Image.FromFile(listBox1.SelectedItem.ToString())

			Else ' Must be the default images in the ImageList
				pbImage.Image = ilDefault.Images(listBox1.SelectedIndex)

			End If
		End Sub


	End Class
End Namespace

File: Professional Image Modifier 1/[C#]-Professional Image Modifier 1/C#/ImageFunctions/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ImageFunctions
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
			
		}




		private void btnLoadNewImage_Click(object sender, EventArgs e)
		{
			DialogResult dr = openFile.ShowDialog();
			if (dr == System.Windows.Forms.DialogResult.OK)
			{
				foreach (string file in openFile.FileNames)
				{
					listBox1.Items.Add(file);
					pbImage.Image = Image.FromFile(file);
				}
			}
		}

		private void listBox1_SelectedIndexChanged(object sender, EventArgs e)
		{
			if (listBox1.SelectedIndex > ilDefault.Images.Count - 1) // Must be added by the user
			{
				pbImage.Image = Image.FromFile(listBox1.SelectedItem.ToString());

			}
			else // Must be the default images in the ImageList
			{
				pbImage.Image = ilDefault.Images[listBox1.SelectedIndex];

			}
		}

		
	}
}
